src/Main/ManaLevelGraph.js

```diff
@@ -0,0 +1,226 @@
+    this.state.bossHealth.series.forEach((series) => {
+      const newSeries = {
+        ...series,
+        data: {},
+      };
+
+      series.data.forEach((item) => {
+        const secIntoFight = Math.floor((item[0] - start) / 1000);
+
+        if (deadBosses.indexOf(series.guid) === -1) {
+          const health = item[1];
+          newSeries.data[secIntoFight] = health;
+
+          if (health === 0) {
+            deadBosses.push(series.guid);
+          }
+        }
+      });
+      bosses.push(newSeries);
+    });
+    const deathsBySecond = {};
+    if (this.state.bossHealth.deaths) {
+      this.state.bossHealth.deaths.forEach((death) => {
+        const secIntoFight = Math.floor((death.timestamp - start) / 1000);
+
+        if (death.targetIsFriendly) {
+          deathsBySecond[secIntoFight] = true;
+        }
+      });
+    }
+
+    const fightDurationSec = Math.ceil((end - start) / 1000);
+    const labels = [];
+    for (let i = 0; i <= fightDurationSec; i += 1) {
+      labels.push(i);
+
+      manaBySecond[i] = manaBySecond[i] !== undefined ? manaBySecond[i] : null;
+      bosses.forEach((series) => {
+        series.data[i] = series.data[i] !== undefined ? series.data[i] : null;
+      });
+      deathsBySecond[i] = deathsBySecond[i] !== undefined ? deathsBySecond[i] : undefined;
+    }
+
+    const data = {
+      labels,
+      datasets: [
+        ...bosses.map((series, index) => ({
+          label: `${series.name} Health`,
+          ...ManaStyles[`Boss-${index}`],
+          ...ManaStyles[`Boss-${series.guid}`],
+          data: Object.keys(series.data).map(key => series.data[key]),
+        })),
+        {
+          label: `Deaths`,
+          ...ManaStyles.Deaths,
+          data: Object.keys(deathsBySecond).map(key => deathsBySecond[key]),
+        },
+        {
+          label: `Mana`,
+          ...ManaStyles.Mana,
+          data: Object.keys(manaBySecond).map(key => manaBySecond[key]),
+        },
+      ],
+    };
+
+    const gridLines = ManaStyles.gridLines;
+
+    const options = {
+      responsive: true,
+      scales: {
+        yAxes: [{
+          gridLines: gridLines,
+          ticks: {
+            callback: (value, index, values) => {
+              return `${value}%`;
+            },
+            min: 0,
+            max: 100,
+            stepSize: 25,
+            fontSize: 14,
+          },
+        }],
+        xAxes: [{
+          gridLines: gridLines,
+          ticks: {
+            callback: (value, index, values) => {
+              if(value%30 === 0)
+                return formatDuration(value);
+              return null;
+            },
+            fontSize: 14,
+          },
+        }],
+      },
+      animation: {
+        duration: 0,
+      },
+      hover: {
+        animationDuration: 0,
+      },
+      responsiveAnimationDuration: 0,
+      tooltips: {
+        enabled: false,
+      },
+      legend: ManaStyles.legend,
+    };
+
+    Chart.plugins.register({
```

Comment:
A better place for this would probably be in the constructor of this class, and even then we risk registering the same plugin multiple times. Also having this in the mana-specific class would encourage copy-pasting this to other modules. If you have any ideas how we could improve that, it would be much appreciated but as long as it doesn't actually lead to problems and so long as it hasn't been duplicated yet it's ok to leave it as is.